### ISSUE TITLE
Negation in Generator and Free Variables

### ISSUE THREAD

**Issue by vale1410** [vale1410/bule#12]
%% %% %%% Problem with negation and _ :
%% val[X], ~map[_,X] :: debug[X].

**Comment by vale1410** [comment]
I guess this should be not allowed. Variables occurring in negated facts in guard need to be bound by positive generators.

**Comment by AbdallahS** [comment]
Agreed. This shouldn't be allowed. One should first project `map` on the second argument before using this pattern.
